src/OpenTelemetry/Metrics/Exemplar/Exemplar.cs

```diff
@@ -20,39 +19,150 @@ namespace OpenTelemetry.Metrics;
 #endif
 public
 #else
-/// <summary>
-/// Represents an Exemplar data.
-/// </summary>
-#pragma warning disable SA1623 // The property's documentation summary text should begin with: `Gets or sets`
 internal
 #endif
     struct Exemplar
 {
+    internal HashSet<string>? ViewDefinedTagKeys;
+
+    private static readonly ReadOnlyFilteredTagCollection Empty = new(excludedKeys: null, Array.Empty<KeyValuePair<string, object?>>(), count: 0);
+    private int tagCount;
+    private KeyValuePair<string, object?>[]? tagStorage;
+    private MetricPointValueStorage valueStorage;
+
     /// <summary>
     /// Gets the timestamp.
     /// </summary>
-    public DateTimeOffset Timestamp { get; internal set; }
+    public DateTimeOffset Timestamp { get; private set; }
 
     /// <summary>
     /// Gets the TraceId.
     /// </summary>
-    public ActivityTraceId? TraceId { get; internal set; }
+    public ActivityTraceId? TraceId { get; private set; }
```

Comment:
I think we dont need to define these a nullable. Similar to logs, if the exemplar is recorded outside of trace context then it would be just `default`. Applies to `SpanId` as well.

Reply:
Updated: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5386/commits/eede4d1317a1f7b9362f3b4ac14d1c9cb87ec01f